scripts/cryptoGen.sh

```diff
@@ -1,29 +1,51 @@
 #! /bin/bash
 
+create_pki() {
+    echo "Creating PKIs folders"
+    mkdir -p "$BASE_DIR/crypto/$1"
+
+    echo "Generating private key for $1"
+    openssl ecparam -name prime256v1 -genkey -noout -out "$BASE_DIR/crypto/$1/$1.key"
+
+    echo "Generate node CSR"
+    openssl req -new -key "$BASE_DIR/crypto/$1/$1.key" -out "$BASE_DIR/crypto/$1/$1.csr" -subj "/C=IL/ST=Haifa/O=BCDB"
+
+    echo "Generate node certificate"
+    openssl x509 -req -in "$BASE_DIR/crypto/$1/$1.csr" -CA "$BASE_DIR/crypto/CA/CA.pem" -CAkey "$BASE_DIR/crypto/CA/CA.key" -CAcreateserial -out "$BASE_DIR/crypto/$1/$1.pem" -days 365 -sha256
+
+}
+
+if [ -z "$1" ]
+  then
+    echo "cryptoGen.sh folder [extra users]"
+    exit 1
+fi
 BASE_DIR=$1
 
-rm -rf "$BASE_DIR/pki"
+echo "$$"
+
+rm -rf "$BASE_DIR/crypto"
 
 echo "Creating PKIs folders"
-mkdir -p "$BASE_DIR/pki/ca"
+mkdir -p "$BASE_DIR/crypto/CA"
 
 echo "Generate root CA private key"
-openssl ecparam -name prime256v1 -genkey -noout -out "$BASE_DIR/pki/ca/rootCA.key"
+openssl ecparam -name prime256v1 -genkey -noout -out "$BASE_DIR/crypto/CA/CA.key"
```

Comment:
Does this script require significant change when we move to cluster? Like one CA, multiple nodes, multiple users?

Reply:
Multiple users already supported. Multiple servers can be added quite easy. In cluster environment it should be splitted to 3 scripts - one for CA generation, one for users and one for nodes.

Reply:
If you will look in `create_pki` function - it creates private key and certificate for some entity - it can be user or node and sign it with CA certificate. We can use it as base for any script that creates crypto for any amount of users/nodes.